### PR TITLE
fix sharing of auras with a closing bracket in name

### LINKED ISSUE #1225: Weakauras not importing via chat caused by special symbols.
**Describe the bug**

Whenever I try to share a weakaura that has special symbols in it's name such as [ ] in case, it displays an error "Requested display does not exist"

**Do you have an error log of what happened?**
 

There are no errors in bug-grabber I use it all the time to analyze this stuff in case shit hits the fan. 

**To Reproduce**
 

**Screenshots**

https://i.gyazo.com/dfeb36d5793c885ea6da74a794b8ff4e.png

**Did you try having WeakAuras as the only enabled addon and everything else (especially something like ElvUI) disabled?**

Yes, base blizz + wa's

**Which version of WeakAuras are you using?**

2.11.6

**Was it working in a previous version? If yes, which was the last good one?**

No

**Additional context**

### PR DESCRIPTION
**Pull request by mrbuds**
# Description
Fixes #1225

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
- [x] Share aura "[hello]"
- [x] Share group "normal name"

### CODE REVIEW

File: WeakAuras/Transmission.lua [hunk 1]
```diff
@@ -145,7 +145,7 @@ local function filterFunc(_, event, msg, player, l, cs, t, flag, channelId, ...)
   local remaining = msg;
   local done;
   repeat
-    local start, finish, characterName, displayName = remaining:find("%[WeakAuras: ([^%s]+) %- ([^%]]+)%]");
+    local start, finish, characterName, displayName = remaining:find("%[WeakAuras: ([^%s]+) %- (.*)%]");
```
Comment (emptyrivers) [inline WeakAuras/Transmission.lua]: This fails on the following chat message:

> Hey guys, look at these great auras: [WeakAuras: EmptyRivers - Aura 1] [WeakAuras: EmptyRivers - Aura 2]

The first link will request an aura named `Aura 1] [WeakAuras: EmptyRivers - Aura 2]`. I'm not entirely sure how the second link will behave, though.

File: WeakAuras/Transmission.lua [hunk 2]
```diff
@@ -546,7 +546,7 @@ local receivedData;
 hooksecurefunc("ChatFrame_OnHyperlinkShow", function(self, link, text, button)
   buttonAnchor:Hide()
   if(link == "weakauras") then
-    local _, _, characterName, displayName = text:find("|Hweakauras|h|cFF8800FF%[([^%s]+) |r|cFF8800FF%- ([^%]]+)%]|h");
+    local _, _, characterName, displayName = text:find("|Hweakauras|h|cFF8800FF%[([^%s]+) |r|cFF8800FF%- (.*)%]|h");
```
Comment (emptyrivers) [inline WeakAuras/Transmission.lua]: This also fails when somebody sends this chat message (for the same reason):
> Hey guys, look at these great auras: [WeakAuras: EmptyRivers - Aura 1] [WeakAuras: EmptyRivers - Aura 2]

**Comment by emptyrivers** [comment]
Well, you might be able to get around the fact that there are pretty much always tricky strings which defeat your patterns, if we can use a hyperlink i.e. `|HWeakAuras:name:displayName|h`. I have no idea if this is something which we are allowed to do or not, though.

File: WeakAuras/Transmission.lua (hunk 1 above)
Comment (mrbuds) [reply WeakAuras/Transmission.lua]: i have always used weakauras links with nothing else than 1 link, i didn't know 2 links in same line was supported tbh

**Comment by mrbuds** [comment]
TotalRp3 read the hyperlink by hooking ChatFrame_OnHyperlinkShow
https://github.com/Ellypse/Total-RP-3/blob/master/totalRP3/modules/ChatLinks/ChatLinks.lua#L228

This part format the chat message
https://github.com/Ellypse/Total-RP-3/blob/master/totalRP3/modules/ChatLinks/ChatLinks.lua#L129

File: WeakAuras/Transmission.lua (hunk 1 above)
Comment (InfusOnWoW) [reply WeakAuras/Transmission.lua]: It also fails if the chat message has a ] in it after the link. So "Guys, install this: [WeakAuras: EmptyRivers - Aura 1]  [].

The "proper" way to do this is to escape any ] in the name of the aura and to properly parse the msg.

**Comment by Stanzilla** [comment]
From the code comments it reads like shift clicking is not supported with that method?

**Comment by mrbuds** [comment]
I fear all fix i could try to make will add another bug for close to no gain at the end .. so i guess it's better if just close this and restore won't fix label on original issue

**Pull request closed without merging**